scripts/get-docker-tag.sh

```diff
@@ -13,22 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-# Helper script to get our docker tag if we're checked out on a branch
-# e.g. 1.4.0.  If we're on master, or if this is a local branch, and we use tag
-# "latest".
-#
-# This lets us stop overwriting gcr.io/config-management-release/nomos:latest
-# with images built on branches.
-#
-# Additional backstory in b/159654901
-#
-# Get the upstream tracking branch.  Note that this needs to work in multiple
-# checkout/fetch environments.  See b/159654901 for the various changes that we
-# tried before settling on this.
-BRANCH=$(git symbolic-ref --short --quiet HEAD);
-if [[ "${BRANCH}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-  echo "${BRANCH}-latest"
+describe=$(git describe --tags --always --dirty --long)
+if [[ "${describe}" == *-dirty ]]; then
+  # Append hash of diff to help enable unique dirty repo states
+  echo "${describe}-$(git diff | sha1sum | head -c 7)"
```

Comment:
FWIW, docker tags have a limit of 128 chars, so this means we have a branch name limit of 128-22 chars. Do we want to have some sort of truncation? Error?

Reply:
I'm pretty sure this will only choose tags (not branches unless the [--all](https://git-scm.com/docs/git-describe#Documentation/git-describe.txt---all) option is provided)

Reply:
After thinking about this a bit more I think there might be value in removing the extra hash and treating `dirty` as a floating tag. This would enable users to make experimental changes and just rebuild the part they want to rebuild.

I'm imagining a use case like:
- User has built images and manifests on a given dirty commit
- User wants to make a change to just the manifests, so updates the manifests and runs `build-manifests`

This way the user can choose to use the previously built docker images without needing to rebuild them for the new manifests. If the user wants to rebuild both then they can still choose to run both `build-images` and `build-manifests`. I suppose they also could have run `retag-images` and `build-manifests` to avoid rebuilding, but that seems a bit less intuitive to me.

@karlkfi WDYT?

Reply:
You can probably get the best of both worlds by using a tombstone that contains the names & tags of the last build images in .env format. That way you can rebuild the manifests without regenerating the image tags.

Reply:
I switched this back to treating `-dirty` as a floating tag, since I expect that to be useful in development workflows. I think we can iterate on this some more after getting feedback.